SwiftCompilerSources/Sources/Optimizer/DataStructures/Worklist.swift

```diff
@@ -0,0 +1,76 @@
+//===--- Worklist.swift ---------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SIL
+
+/// A utility for processing entities in a worklist.
+///
+/// A `Worklist` is basically a combination of a stack and a set.
+/// It can be used for typical worklist-processing algorithms.
+///
```

Comment:
@eeckstein In addition to the "standard" push/pop worklist I sometimes find it convenient to process a worklist in breadth-first order without popping, using an advancing iterator. That's very useful when you need a list of all the visited nodes as a result of processing the worklist.

Reply:
Yeah, we can add that when we need it